drivers/platform/xilinx/gpio.c

```diff
@@ -382,12 +382,35 @@ int32_t gpio_set_value(struct gpio_desc *desc,
 int32_t gpio_get_value(struct gpio_desc *desc,
 		       uint8_t *value)
 {
-	if (desc) {
-		// Unused variable - fix compiler warning
-	}
+	if (!desc)
+		return FAILURE;
 
-	if (value) {
-		// Unused variable - fix compiler warning
+	struct xil_gpio_desc	*extra;
+#ifdef XGPIO_H
+	uint8_t pin = desc->number;
```

Comment:
Nitpick: I think is better if we  declare the variables at the begging of the function